ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Basics.hs

```diff
@@ -798,15 +789,43 @@ data ApplyMapKind' :: MapKind' -> Type -> Type -> Type where
   ApplyQueryAllMK  ::                ApplyMapKind' QueryMK' k v
   ApplyQuerySomeMK :: !(Keys k v) -> ApplyMapKind' QueryMK' k v
 
-emptyAppliedMK :: (Ord k, Eq v) => SMapKind mk -> ApplyMapKind' mk k v
-emptyAppliedMK = \case
-    SEmptyMK    -> ApplyEmptyMK
-    SKeysMK     -> ApplyKeysMK      mempty
-    SValuesMK   -> ApplyValuesMK    mempty
-    STrackingMK -> ApplyTrackingMK  mempty mempty
-    SDiffMK     -> ApplyDiffMK      mempty
-    SSeqDiffMK  -> ApplySeqDiffMK   empty
-    SQueryMK    -> ApplyQuerySomeMK mempty
+class IsApplyMapKind mk where
```

Comment:
I know this wasn't documented before, but perhaps we should use this opportunity to explain what `IsApplyMapKind` is. Eg:

- Why do we need it?
- What does it model?
- What does it do?